packages/turf-clusters-distance/test.js

```diff
@@ -0,0 +1,144 @@
+    t.throws(() => clustersDistance(points, 'foo'), /Invalid maxDistance/);
+    t.throws(() => clustersDistance(points, 1, 'nanometers'), /units is invalid/);
+    t.throws(() => clustersDistance(points, 1, null, 0), /Invalid minPoints/);
+    t.throws(() => clustersDistance(points, 1, 'miles', 'baz'), /Invalid minPoints/);
+    t.end();
+});
+
+test('clusters -- prevent input mutation', t => {
+    clustersDistance(points, 2, 'kilometers', 1);
+    t.true(points.features[0].properties.cluster === undefined, 'cluster properties should be undefined');
+    t.end();
+});
+
+test('clusters -- translate properties', t => {
+    t.equal(clustersDistance(points, 2, 'kilometers', 1).features[0].properties.foo, 'bar');
+    t.end();
+});
+
+// style result
+function colorize(clustered) {
+    const clusters = new Set();
+    propEach(clustered, ({cluster}) => clusters.add(cluster));
+    const count = clusters.size;
+    const colours = chromatism.adjacent(360 / count, count, '#0000FF').hex;
+    const points = [];
+
+    featureEach(clustered, function (point) {
+        switch (point.properties.dbscan) {
+        case 'core':
+        case 'edge': {
+            const coreColor = colours[point.properties.cluster];
+            const edgeColor = chromatism.brightness(-20, colours[point.properties.cluster]).hex;
+            point.properties['marker-color'] = (point.properties.dbscan === 'core') ? coreColor : edgeColor;
+            point.properties['marker-size'] = 'small';
+            points.push(point);
+            break;
+        }
+        case 'noise': {
+            point.properties['marker-color'] = '#AEAEAE';
+            point.properties['marker-symbol'] = 'circle-stroked';
+            point.properties['marker-size'] = 'medium';
+            points.push(point);
+        }
+        }
+    });
+    // Create Centroid from cluster property
+    const properties = {
+        'marker-symbol': 'star',
+        'marker-size': 'large'
+    };
+    const centroids = centroidFromProperty(points, 'cluster', properties);
+    featureEach(centroids, point => {
+        const color = chromatism.brightness(-15, colours[point.properties.cluster]).hex;
+        point.properties['marker-color'] = color;
+        points.push(point);
+    });
+
+    return featureCollection(points);
+}
+
+/**
+ * Basic implementation of creating centroids based on a single Property
+ * Can be expanded with very complex property combination
+ *
+ * @private
+ * @param {FeatureCollection|Feature[]} geojson GeoJSON
+ * @param {string} property Name of property of GeoJSON Properties to bin each feature - This property will be translated to each centroid's properties
+ * @param {*} properties Properties to translate down to each centroid (2nd priority)
+ * @returns {FeatureCollection<Point>} centroids
+ * @example
+ * var centroids = centroidFromProperty(points, 'cluster');
+ */
+function centroidFromProperty(geojson, property, properties) {
```

Comment:
Yes! :) only for `test.js` This where this module would be used... we wouldn't apply this directly in the modules, but for display purposes.

Also this can be applied against Polygons or any Geometry Types, finding the "centroid" of stuff.